Integer scaling using buffer_scale

arnor-nolen opened the pull request:
Hey, I like your bar quite a lot, so I decided to contribute. 

I am personally using 1.25 and 1.5 scale values on my monitors and I have noticed that your bar doesn't support scaling, so the bar looks very blurry/pixelated. If you know how scaling is organized in Wayland, there's two ways of dealing with scaling: so-called "integer scaling" and "fractional scaling". Fractional scaling is quite new, so I decided to implement integer scaling through an ability to supply `buffer_scale` value to the compositor. This allows you to render surfaces at (for example) 2x scale and the compositor will do the rest of the job by scaling it down to 1.25x or 1.5x. This is not ideal, but this is how scaling on Wayland was done before fractional scaling was introduced.

I might also create a PR that uses fractional scaling, but integer scaling is a good starting point.

Not sure if this is supposed to be a patch or PR, feel free to propose your solution.

Changelog:
* Added an ability to specify `buffer_scale`. This allows to use integer scaling for HiDPI displays.

kolunmi commented:
Hello! Thanks so much for the pr! Could you add a `-scale` option to the help text and accept it as a command line option so I can merge this?

arnor-nolen commented:
Added `-scale` as an option you can pass, also added explanations to both help message and README file. If you don't like my wordings, you can change it as you like. 

Here's an example of using 1.5 scaling monitor.
Before patching (or running with `dwlb -scale 1`):
![image](https://user-images.githubusercontent.com/18191106/235858870-7d9f9364-099c-4503-89fd-4708fb8d106d.png)
After patching (with `dwlb -scale 2`):
![image](https://user-images.githubusercontent.com/18191106/235859257-51ba094a-5eb4-4e14-bcf5-6ec0fe6daf1d.png)

As you can see, size of the image did not change, but everything is rendered at higher detail.

kolunmi commented:
Looks good, thank you!
